server/api/units/unit-metrics.controller.js

```
@@ -200,51 +143,120 @@ export function runQuery(req, res) {
       };
 
       // total data
-      const apparatusBuckets = _.get(esRes, 'aggregations.apparatus["agg_terms_apparatus.unit_id"].buckets');
+      const apparatusBuckets = esRes.aggregations.apparatus['agg_terms_apparatus.unit_id'].buckets;
       let totalMetrics = {};
       _.forEach(apparatusBuckets, b => {
-        totalMetrics[b.key] = getMetrics(metrics, b);
+        totalMetrics[b.key] = getMetricsFromBucket(metrics, b);
       });
       const rankedMetrics = rankBuckets(totalMetrics);
       api_response.total_data = rankedMetrics[unitId];
 
       // grouped data
-      const categoryBuckets = _.get(esRes, 'aggregations["agg_terms_description.category"]buckets');
```

can you put these _.get guards back in?  I think its possible that es doesn't set the aggregations, which would cause the server to crash on the next line when trying to access a field that doesn't exist

---

@chopchop505 These are restored as of 0d118d74ea1306882e9bbf53cc5358f57833d4e0. I thought about wrapping those chunks with if checks to avoid the _.get() calls, which would fail silently if syntax errors slipped through, but the code is already well structured for them and I don't want to mess with it any more than I already have. I have mixed feelings about using _.get() since it lets any kind of error in the code slip by unnoticed... but maybe that's a good thing on the server for stability. You probably know better with this stuff than I do.